src/raw_node.rs

```diff
@@ -563,15 +563,17 @@ impl<T: Storage> RawNode<T> {
     /// valid after ready being persisted.
     pub fn on_persist_ready(&mut self, number: u64) {
         let (mut index, mut term) = (0, 0);
+        let mut snap_index = 0;
         while let Some(record) = self.records.front() {
             if record.number > number {
                 break;
             }
             let mut record = self.records.pop_front().unwrap();
 
-            if let Some((i, t)) = record.snapshot {
-                index = i;
-                term = t;
+            if let Some((i, _)) = record.snapshot {
```

Comment:
Is it possible that some entries follow one snapshot in one ready?

Reply:
Yes, it happens when the snapshot has up-to-date data and the leader is changed.
If a ready has a snapshot, the entries of the readies before this one must have smaller indexes so they don't need to call `on_persist_entries`.